src/containerapp/azext_containerapp/containerapp_env_decorator.py

```diff
@@ -149,6 +148,21 @@ def set_up_custom_domain_configuration(self):
                 }
             self.managed_env_def["properties"]["customDomainConfiguration"] = custom_domain
 
+    def set_up_peer_to_peer_encryption(self):
+        is_p2p_encryption_enabled = self.get_argument_p2p_encryption_enabled()
+        is_mtls_enabled = self.get_argument_mtls_enabled()
+        logger.info("mtls: p2p_encryption: %s, mtls: %s", is_p2p_encryption_enabled, is_mtls_enabled)
+
+        if is_p2p_encryption_enabled is not None:
+            safe_set(self.managed_env_def, "properties", "peerTrafficConfiguration", "encryption", "enabled", value=is_p2p_encryption_enabled)
+            if is_p2p_encryption_enabled is False:  # keep the two properties consistent.
+                safe_set(self.managed_env_def, "properties", "peerAuthentication", "mtls", "enabled", value=False)
```

Comment:
The logic here means if `--enable-peer-to-peer-encryption False`, set:
`properties.peerTrafficConfiguration.encryption.enabled` as `False`
`properties.peerAuthentication.mtls.enabled` as `False`

But what if `--enable-peer-to-peer-encryption False --enable-mtls`?
`properties.peerAuthentication.mtls.enabled` will be set as `False`
Then in line 162 be set to True.
But 
`properties.peerTrafficConfiguration.encryption.enabled` will be set as `True` at line 164

Reply:
The logic you add to `keep the two properties consistent.` make some input not make sense

Reply:
I think you should write a warning logs to customer if the customer provided value is overwritten, or fail the command in validation logic.